providers/linux/host_fqdn_integration_linux_test.go

```diff
@@ -39,145 +37,96 @@ const (
-			cf: container.Config{
-				Hostname:     wantHostname,
-				Domainname:   wantDomain,
-				AttachStderr: testing.Verbose(),
-				AttachStdout: testing.Verbose(),
-				WorkingDir:   "/usr/src/elastic/go-sysinfo",
-				Image:        image,
-				Cmd: []string{
-					"go", "test", "-v",
-					"-tags", "integration,docker",
-					"-run", "^TestHost_FQDN_set$",
-					"./providers/linux",
-				},
-				Tty: false,
+			name:       "TestHost_FQDN_set_hostname+domainname",
+			Hostname:   wantHostname,
+			Domainname: wantDomain,
+			Cmd: []string{
+				"go", "test", "-v",
+				"-tags", "integration,docker",
+				"-run", "^TestHost_FQDN_set$",
+				"./providers/linux",
 			},
 		},
 		{
-			name: "TestHost_FQDN_set_hostname_only",
-			cf: container.Config{
-				Hostname:     wantFQDN,
-				AttachStderr: testing.Verbose(),
-				AttachStdout: testing.Verbose(),
-				WorkingDir:   "/usr/src/elastic/go-sysinfo",
-				Image:        image,
-				Cmd: []string{
-					"go", "test", "-v",
-					"-tags", "integration,docker",
-					"-run", "^TestHost_FQDN_set$",
-					"./providers/linux",
-				},
-				Tty: false,
+			name:     "TestHost_FQDN_set_hostname_only",
+			Hostname: wantFQDN,
+			Cmd: []string{
+				"go", "test", "-v",
+				"-tags", "integration,docker",
+				"-run", "^TestHost_FQDN_set$",
+				"./providers/linux",
 			},
 		},
 		{
 			name: "TestHost_FQDN_not_set",
-			cf: container.Config{
-				AttachStderr: testing.Verbose(),
-				AttachStdout: testing.Verbose(),
-				WorkingDir:   "/usr/src/elastic/go-sysinfo",
-				Image:        image,
-				Cmd: []string{
-					"go", "test", "-v", "-count", "1",
-					"-tags", "integration,docker",
-					"-run", "^TestHost_FQDN_not_set$",
-					"./providers/linux",
-				},
-				Tty: false,
+			Cmd: []string{
+				"go", "test", "-v", "-count", "1",
+				"-tags", "integration,docker",
+				"-run", "^TestHost_FQDN_not_set$",
+				"./providers/linux",
 			},
 		},
 	}
 
-	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
-	if err != nil {
-		t.Fatalf("failed to create docker client: %v", err)
-	}
-	defer cli.Close()
-
 	for _, tc := range tcs {
 		t.Run(tc.name, func(t *testing.T) {
-			runOnDocker(t, cli, &tc.cf)
+			dockerRun(t, tc.Hostname, tc.Domainname, tc.Cmd)
 		})
 	}
 }
 
-func runOnDocker(t *testing.T, cli *client.Client, cf *container.Config) {
-	ctx := context.Background()
+// dockerRun executes the given command inside the golang Docker container.
+// It will set the container's hostname and domain according to the given
+// values.
+//
+// The container's stdout and stderr are passed through. The test will fail
+// if docker run returns a non-zero exit code.
+func dockerRun(t *testing.T, hostname, domain string, command []string) {
+	t.Helper()
 
-	pwd, err := os.Getwd()
-	if err != nil {
-		t.Fatalf("could not get current directory: %v", err)
+	if _, err := execabs.LookPath("docker"); err != nil {
+		t.Skipf("Skipping because docker was not found: %v", err)
 	}
```

Comment:
Hoist this to the outer loop? If it's not here once, it's likely not here at all.